From 21e0c16a85c2e028a963703511d750d1751bc254 Mon Sep 17 00:00:00 2001
From: Alexey Brodkin <abrodkin@synopsys.com>
Date: Wed, 13 Jul 2016 23:46:38 +0200
Subject: [PATCH] Fix a build issue with uClibc-ng

uClibc-ng has clock_nanosleep() if built with UCLIBC_HAS_ADVANCED_REALTIME,
  conflicting with emulation function.

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
[Romain:
    rebase on v1.0
    remove uClibc legacy support, uClibc-ng provide utmpx.h]
Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 src/cyclictest/cyclictest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cyclictest/cyclictest.c b/src/cyclictest/cyclictest.c
index 00e5f3d..92fa3dd 100644
--- a/src/cyclictest/cyclictest.c
+++ b/src/cyclictest/cyclictest.c
@@ -58,7 +58,7 @@
 #define gettid() syscall(__NR_gettid)
 #define sigev_notify_thread_id _sigev_un._tid
 
-#ifdef __UCLIBC__
+#if defined(__UCLIBC__) && !defined(__UCLIBC_HAS_ADVANCED_REALTIME__)
 #define MAKE_PROCESS_CPUCLOCK(pid, clock) \
 	((~(clockid_t) (pid) << 3) | (clockid_t) (clock))
 #define CPUCLOCK_SCHED          2
-- 
2.5.5

